samples/v1beta1/buildstrategy/multiarch-native-buildah/buildstrategy_multiarch_native_buildah_cr.yaml

```diff
@@ -0,0 +1,545 @@
+            elif [ "${status}" == "parse_registries_block" ]; then
+              registriesBlock="${registriesBlock}'${arg}', "
+            elif [ "${status}" == "parse_registries_insecure" ]; then
+              registriesInsecure="${registriesInsecure}'${arg}', "
+            elif [ "${status}" == "parse_registries_search" ]; then
+              registriesSearch="${registriesSearch}'${arg}', "
+            else
+              echo "[ERROR] Invalid usage"
+              exit 1
+            fi
+          done
+
+          # Verify the existence of the context directory
+          if [ ! -d "${context}" ]; then
+            echo -e "[ERROR] The context directory '${context}' does not exist."
+            echo -n "ContextDirNotFound" > '$(results.shp-error-reason.path)'
+            echo -n "The context directory '${context}' does not exist." > '$(results.shp-error-message.path)'
+            exit 1
+          fi
+          cd "${context}"
+
+          # Verify the existence of the Dockerfile
+          if [ ! -f "${dockerfile}" ]; then
+            echo -e "[ERROR] The Dockerfile '${dockerfile}' does not exist."
+            echo -n "DockerfileNotFound" > '$(results.shp-error-reason.path)'
+            echo -n "The Dockerfile '${dockerfile}' does not exist." > '$(results.shp-error-message.path)'
+            exit 1
+          fi
+          echo "[INFO] Creating registries config file..."
+          touch /tmp/registries.conf
+          if [ "${registriesSearch}" != "" ]; then
+            cat <<EOF >>/tmp/registries.conf
+          [registries.search]
+          registries = [${registriesSearch::-2}]
+          EOF
+          fi
+          if [ "${registriesInsecure}" != "" ]; then
+            cat <<EOF >>/tmp/registries.conf
+          [registries.insecure]
+          registries = [${registriesInsecure::-2}]
+          EOF
+          fi
+          if [ "${registriesBlock}" != "" ]; then
+            cat <<EOF >>/tmp/registries.conf
+          [registries.block]
+          registries = [${registriesBlock::-2}]
+          EOF
+          fi
+
+          if [ "${#runtime_stage_from_image}" -gt 0 ]; then
+            echo "[INFO] Replacing the runtime stage image to extend from"
+            buildArgs+=("--build-context" "$(tac "${dockerfile}" | grep -m1 -i -E '^[ ]*FROM' | \
+              sed -n '0,/p/s/^[ ]*from[ ]\+\([^ ]*\)[ ]*\(as.*$\)\{0,1\}$/\1/Ip')=${runtime_stage_from_image}")
+          fi
+          # TODO: what image including kubectl can we use?
+          echo "[INFO] Downloading kubectl..."
+          kube_version=$(curl --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+            "https:${KUBERNETES_PORT#tcp:}/version" | sed -n 's/^.*gitVersion.*"v\(.*\)\+".*$/\1/p')
+          arch=$(uname -m | sed 's/x86_64/amd64/;s/aarch64/arm64/')
+          curl -L -o /usr/local/bin/kubectl \
+            "https://storage.googleapis.com/kubernetes-release/release/v${kube_version}/bin/linux/${arch}/kubectl"
```

Comment:
I ran into this failing when kubernetes version was reported by the server as `v1.28.9+416ecaf`. Google wants only `v1.28.9` it seems.

Reply:
@dorzel good point, I just submitted a change that should cover that case too.

Reply:
https://github.com/shipwright-io/build/pull/1634/files#diff-d244d2f9d7466893a1cfb33fef3b6c411fcae737e967fed9571c9ac30f3ad652R158